### PR TITLE
config: remove allow-auto-random config option

### LINKED ISSUE #17882: Move `auto_random` out of experimental features
## Development Task

Since the attribute `auto_random` is stable now, we can move the option `allow-auto-random` out of the [experimental] section in the configuration file.

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?
Close #17882

Problem Summary: Since the attribute `auto_random` is stable now, we can move the option `allow-auto-random` out of the `[experimental]` section in configuration file.

### What is changed and how it works?

What's Changed:

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 
- The column attribute `auto_random` gets stable and it can be used without the config option `allow-auto-random`.

### CODE REVIEW

File: config/config.toml.example [hunk 1]
```diff
@@ -87,6 +87,9 @@ split-region-max-num = 1000
 # the primary key cannot be dropped.
 alter-primary-key = false
 
+# enable column attribute `auto_random` to be defined on the primary key column.
+allow-auto-random = false
```
Comment (bb7133) [inline config/config.toml.example:91]: Should we remove this configuration?

File: ddl/serial_test.go [hunk 2]
```diff
@@ -838,6 +835,13 @@ func (s *testSerialSuite) TestAutoRandom(c *C) {
 	assertPKIsNotHandle("create table t (a bigint auto_random(3), b int, primary key (a, b))", "a")
 	assertPKIsNotHandle("create table t (a int auto_random(3), b int, c char, primary key (a, c))", "a")
 
+	// PKIsNotHandle: table is created when alter-primary-key = true.
+	testutil.ConfigTestUtils.EnableAlterPrimaryKeyConfig()
```
Comment (zimulala) [inline ddl/serial_test.go:839]: If `EnableAlterPrimaryKeyConfig` is only used here(test). Could we use `config.GetGlobalConfig().AlterPrimaryKey = true` directly?

File: meta/autoid/errors.go [hunk 3]
```diff
@@ -30,9 +30,7 @@ var (
 
 const (
 	// AutoRandomPKisNotHandleErrMsg indicates the auto_random column attribute is defined on a non-primary key column, or the table's primary key is not a single integer column.
-	AutoRandomPKisNotHandleErrMsg = "column %s is not the single integer primary key, or alter-primary-key is enabled"
-	// AutoRandomExperimentalDisabledErrMsg is reported when the experimental option allow-auto-random is not enabled.
-	AutoRandomExperimentalDisabledErrMsg = "auto_random is an experimental feature, which can only be used when allow-auto-random is enabled. This can be changed in the configuration."
+	AutoRandomPKisNotHandleErrMsg = "column %s is not the integer primary key, or table is created with alter-primary-key enabled"
```
Comment (AilinKid) [inline meta/autoid/errors.go:33]: is there any place checked with this `table is created with alter-primary-key enabled`?

Comment (tangenta) [reply meta/autoid/errors.go:33]: https://github.com/pingcap/tidb/blob/2bb1ff5a1efaec0cb38b575309091254df19d0b3/ddl/serial_test.go#L843

File: util/testutil/testutil.go [hunk 4]
```diff
@@ -324,21 +324,18 @@ type autoRandom struct {
 	originAlterPrimaryKey bool
 }
 
-// SetupAutoRandomTestConfig set alter-primary-key to false, and set allow-auto-random to true and save their origin values.
+// SetupAutoRandomTestConfig set alter-primary-key to false and save its origin values.
 // This method should only be used for the tests in SerialSuite.
 func (a *autoRandom) SetupAutoRandomTestConfig() {
```
Comment (zimulala) [inline util/testutil/testutil.go:329]: Do we need to update this function name? This function doesn't handle `AutoRandom` related configurations now.

File: meta/autoid/errors.go (hunk 3 above)
Comment (AilinKid) [reply meta/autoid/errors.go:33]: got, 
there is not explicit AlterPrimaryKey judgement, when this is on, pk is not handle indeed.

File: util/testutil/testutil.go (hunk 4 above)
Comment (tangenta) [reply util/testutil/testutil.go:329]: This is only used in `auto_random` tests and it does set up the config. I think no need to update.

**Comment by tangenta** [comment]
/common-test
```
[2020-06-10T05:29:46.910Z] ./t/drop.test: ok! 19 test cases passed, take time 0.253035931 s

[2020-06-10T05:29:47.166Z] time="2020-06-10T13:29:46+08:00" level=info msg="run test [drop] ok"

[2020-06-10T05:29:47.166Z] time="2020-06-10T13:29:46+08:00" level=warning msg="Create new db&{0 {root:@tcp(127.0.0.1:4001)/test?strict=true&time_zone=%27Asia%2FShanghai%27 0x10a1110} 0 {0 0} [0xc0006e8900] map[] 0 1 0xc0006cea80 0xc0005f1e00 false map[0xc0006e8900:map[0xc0006e8900:true]] map[] 0 0 0 <nil> 0 0 0 0x48c430}"

[2020-06-10T05:29:47.166Z] time="2020-06-10T13:29:46+08:00" level=fatal msg="run test [insert_select] err: sql:INSERT INTO t1 VALUES (1,1),(2,2),(10,10);: run \"INSERT INTO t1 VALUES (1,1),(2,2),(10,10);\" at line 134 err Error 1146: Table 'test.t1' doesn't exist"
```

**Comment by tangenta** [comment]
/run-unit-test
Data race:
```
[2020-06-10T05:34:50.397Z] WARNING: DATA RACE

[2020-06-10T05:34:50.397Z] Write at 0x00c000171158 by goroutine 255:

[2020-06-10T05:34:50.397Z]   github.com/pingcap/tidb/session_test.(*testSessionSerialSuite).TestStatementCountLimit()

[2020-06-10T05:34:50.397Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/session_test.go:2456 +0x1c9

[2020-06-10T05:34:50.397Z]   github.com/pingcap/tidb/session_test.(*testSessionSerialSuite).TestStatementCountLimit()

[2020-06-10T05:34:50.397Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/session_test.go:2454 +0x12d

[2020-06-10T05:34:50.397Z]   runtime.call32()

[2020-06-10T05:34:50.397Z]       /usr/local/go/src/runtime/asm_amd64.s:539 +0x3a

[2020-06-10T05:34:50.397Z]   reflect.Value.Call()

[2020-06-10T05:34:50.397Z]       /usr/local/go/src/reflect/value.go:321 +0xd3

[2020-06-10T05:34:50.397Z]   github.com/pingcap/check.(*suiteRunner).forkTest.func1()

[2020-06-10T05:34:50.397Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0x9aa

[2020-06-10T05:34:50.397Z]   github.com/pingcap/check.(*suiteRunner).forkCall.func1()

[2020-06-10T05:34:50.397Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x113

[2020-06-10T05:34:50.397Z] 

[2020-06-10T05:34:50.397Z] Previous read at 0x00c000171158 by goroutine 568:

[2020-06-10T05:34:50.397Z]   github.com/pingcap/tidb/session.checkStmtLimit()

[2020-06-10T05:34:50.397Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/tidb.go:247 +0xe7

[2020-06-10T05:34:50.397Z]   github.com/pingcap/tidb/session.finishStmt()

[2020-06-10T05:34:50.397Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/tidb.go:211 +0x133

[2020-06-10T05:34:50.398Z]   github.com/pingcap/tidb/session.runStmtWrap()

[2020-06-10T05:34:50.398Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/session.go:1224 +0x568

[2020-06-10T05:34:50.398Z]   github.com/pingcap/tidb/session.(*session).ExecuteStmt()

[2020-06-10T05:34:50.398Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/session.go:1164 +0xa6f

[2020-06-10T05:34:50.398Z]   github.com/pingcap/tidb/session.(*session).Execute()

[2020-06-10T05:34:50.398Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/session.go:1077 +0x3ee

[2020-06-10T05:34:50.398Z]   github.com/pingcap/tidb/ddl/util.RemoveFromGCDeleteRange()

[2020-06-10T05:34:50.398Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/util/util.go:121 +0x18f

[2020-06-10T05:34:50.398Z]   github.com/pingcap/tidb/ddl/util.CompleteDeleteRange()

[2020-06-10T05:34:50.398Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/util/util.go:115 +0x227

[2020-06-10T05:34:50.398Z]   github.com/pingcap/tidb/ddl.(*delRange).doTask()

[2020-06-10T05:34:50.398Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/delete_range.go:230 +0x611

[2020-06-10T05:34:50.398Z]   github.com/pingcap/tidb/ddl.(*delRange).doDelRangeWork()

[2020-06-10T05:34:50.398Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/delete_range.go:184 +0x54c

[2020-06-10T05:34:50.398Z]   github.com/pingcap/tidb/ddl.(*delRange).startEmulator()

[2020-06-10T05:34:50.398Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/delete_range.go:148 +0x1db
```

**Pull request merged**